README.md

```diff
@@ -98,101 +70,179 @@ Blaze4J bridges Java and native C++ for **fast, standards-compliant JSON Schema
 └── CMakeLists.txt     # CMake configuration
 
 ```
-
 ---
 
-## Dependencies
+## Installation
 
-- Java Development Kit (JDK) 11 or higher
-- CMake 3.10 or higher
-- C++ compiler with C++11 support
-- Blaze library (included as a Git submodule)
-
----
-
-## Build Instructions
+**Maven**
+- Add the following dependency to your `pom.xml`:
+```java
+<dependency>
+    <groupId>io.github.madhavdhatrak</groupId>
+    <artifactId>blaze4j</artifactId>
+    <version>0.0.1</version>
+</dependency>
 
-### 1. Initialize Blaze Submodule
-```
-git submodule add https://github.com/sourcemeta/blaze.git deps/blaze
 ```
 
-### 2. Build for Windows
+**Gradle**
+- Add this to your `build.gradle`:
 ```java
-mkdir build
-cmake -Bbuild -G "Visual Studio 17 2022" -DCMAKE_BUILD_TYPE=Release
-cmake --build build --config Release
+implementation group: 'io.github.madhavdhatrak', name: 'blaze4j', version: '0.0.1'
 ```
-### 3. Build for Linux
+
+ **Make Sure to have pom.xml Like this**
 ```java
-mkdir -p build-linux
-cd build-linux
-cmake .. -DCMAKE_BUILD_TYPE=Release
-make
+<properties>
+    <maven.compiler.source>21</maven.compiler.source>
+    <maven.compiler.target>21</maven.compiler.target>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+</properties>
+
+<dependencies>
+    <dependency>
+        <groupId>io.github.madhavdhatrak</groupId>
+        <artifactId>blaze4j</artifactId>
+        <version>0.0.1</version>
+    </dependency>
+</dependencies>
+
+<build>
+    <plugins>
+        <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-compiler-plugin</artifactId>
+            <version>3.11.0</version>
+            <configuration>
+                <compilerArgs>
+                    <arg>--enable-preview</arg>
+                </compilerArgs>
+            </configuration>
+        </plugin>
+        <plugin>
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>exec-maven-plugin</artifactId>
```

Comment:
Not every user will need this plugin. For example if you use spring boot, then you don't need the exec-maven-plugin. I suggest doing the following:

 * include only the most minimal code snippets in the README that everyone has to be aware of
 *  have a separate `blaze4j-examples` repository that has complete working examples, with all details, so that people can just clone and run it. It could include a hello-world example (plain old java application) and a spring boot example (and maybe more in the future).